### ISSUE TITLE
SNOW-1471194: Adding an option `--use-defaults` with add connection

### ISSUE THREAD

**Issue by kameshsampath** [snowflakedb/snowflake-cli#1178]
### Description

As a developer using snow cli, I would like to have an option like `--use-defaults` that will allow or make `snow connection add` use defaults for optional parameters, given the mandatory ones are passed through CLI 

e.g.
```shell
snow connection add --connection-name cheatsheets --account account-id --user user --password 'password' 
```

IIUC these are the only mandatory parameters for a connection, when I provide them the `connnection add` will prompt for the other optional parameters, in those cases I wish the developer can override the prompts like with an option like `--use-defaults` which will just create or add connection configuration with only options passed by user.

e.g.

```shell
snow connection add --connection-name cheatsheets --account account-id --user user --password 'password'   --use-defaults
```

### Context

When adding new connections using `snow connection add`

**Comment by sfc-gh-turbaszek** [comment]
Set of required input depends on authorization method. That said I think we can alter the logic to:
1. Do not prompt for additional values if one provides:
   2.  --user --account --password 
   3. --user --account --authenticator
2. Prompt for missing input in any other case

WDYT @kameshsampath @sfc-gh-pczajka ?

**Comment by kameshsampath** [comment]
yeah we should prompt for mandatory ones for rest we can leave to defaults when the user says `--use-defaults`, if user does not specify `--use-defaults` then we resort to existing flow of prompting. how does sound?

**Comment by sfc-gh-turbaszek** [comment]
I'm not convinced if we need `--use-defaults`. The usages can be imho split into two:
1. Flag-driven, when user provides input via flag, in this case we we only prompt if configuration is incomplete. We assume user using flags is aware of other possible configuration options.
2. Interactive when user adds connection in a "wizard" mode where we prompt for everything

**Comment by kameshsampath** [comment]
If we don't want to use `--use-defaults` then we need to be assuming something like,

- when user provides parameters for all required fields then don't prompt for optional ones
- if user does not pass parameters then use wizard mode

Point 1 assumes user knows what he wants to set and what not to. My thought of using `--use-defaults` was more of an indicator and double confirmation from user to use defaults for others.

**Comment by sfc-gh-astus** [comment]
We added `--no-interactive` flag.
